src/kili/services/plugins/upload.py

```diff
@@ -0,0 +1,95 @@
+"""
+Class to upload a plugin
+"""
+
+from pathlib import Path
+from typing import Optional
+
+from kili.authentication import KiliAuth
+from kili.graphql.operations.plugins.mutations import (
+    GQL_CREATE_PLUGIN_RUNNER,
+    GQL_GET_PLUGIN_UPLOAD_URL,
+)
+from kili.helpers import check_file_is_py, format_result
+from kili.utils import bucket
+
+
+class PluginUploader:  # pylint: disable=too-few-public-methods
+    """
+    Class to upload a plugin
+    """
+
+    def __init__(
+        self, auth: KiliAuth, file_path: str, plugin_name: Optional[str], verbose: bool
+    ) -> None:
+        self.auth = auth
+        self.file_path = Path(file_path)
+        self.plugin_name = plugin_name or self.file_path.name
+        self.verbose = verbose
+
+    def _retrieve_script(self) -> str:
+        """
+        Retrieve script from file_path and execute it
+        to prevent an upload with indentation errors
+        """
+        if not check_file_is_py(self.file_path, self.verbose):
+            raise ValueError("Wrong file format. ")
+
+        path = Path(self.file_path)
+
+        with path.open("r", encoding="utf-8") as file:
+            source_code = file.read()
+
+        file.close()
```

Comment:
by using the context manager approach (`with open(...) as file`), the file get automatically closed when needed. No need of this line

Reply:
That is what I thought but I was not 100% sure. I'll delete the line